examples/Reranking_search_results_with_cross-encoders.ipynb

```diff
@@ -0,0 +1,805 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "7f30b8b2",
+   "metadata": {},
+   "source": [
+    "# Reranking search results with cross-encoders\n",
+    "\n",
+    "This notebook takes you through examples of using a cross-encoder to re-rank search results.\n",
+    "\n",
+    "This is a common use case with our customers, where you've implemented semantic search using embeddings (produced using a [Bi-encoder](https://www.sbert.net/examples/applications/retrieve_rerank/README.html#retrieval-bi-encoder)) but the results are not exactly what you need. A possible cause is that there is some business rule you can use to rerank the documents such as how recent or how popular a document is. \n",
+    "\n",
```

Comment:
I think it's worth adding a paragraph to introduce what a cross-encoder is.

Here is a suggestion:

"
Consider a search task with D documents and Q queries.

The brute force approach of computing every pairwise relevance is expensive; its cost scales as `D * Q`. This is known as cross-encoding.

A faster approach is embeddings-based search, in which an embedding is computed once for each document and query; its cost scales as `D + Q`. The embeddings are then re-used multiple times to cheaply compute cosine similarity pairs. This is known as bi-encoding.

Although embeddings-based search is faster, the quality can be worse. To get the best of both, one common approach is to use embeddings (bi-encoder) to cheaply identify top candidates, and then use GPT (cross-encoder) to expensively re-rank those top candidates. The cost of this hybrid approach scales as `(D + Q) * cost of embedding + (N * Q) * cost of re-ranking`, where N is the number of candidates re-ranked.

Reply:
I've added this example, though there is some overlap with the second and third paragraphs. Open to your thoughts, we can replace with the above but I want to retain some of the simplicity of the first couple of paragraphs.